llvm/lib/CodeGen/LiveDebugValues/VarLocBasedImpl.cpp

```diff
@@ -1890,20 +1889,29 @@ bool VarLocBasedLDV::ExtendRanges(MachineFunction &MF, TargetPassConfig *TPC) {
 
   // Only in the case of entry MBB collect DBG_VALUEs representing
   // function parameters in order to generate debug entry values for them.
+  SmallVector<MachineInstr *, 8> AsyncDbgValues;
   MachineBasicBlock &First_MBB = *(MF.begin());
   for (auto &MI : First_MBB) {
     collectRegDefs(MI, DefinedRegs, TRI);
     if (MI.isDebugValue()) {
       // In Swift async functions entry values are preferred, since they
       // can be evaluated in both live frames and virtual backtraces.
-      if (isSwiftAsyncContext(MI))
+      if (isSwiftAsyncContext(MI)) {
         MI.getOperand(3).setMetadata(DIExpression::prepend(
             MI.getDebugExpression(), DIExpression::EntryValue));
-      else
+        AsyncDbgValues.push_back(&MI);
+      } else
         recordEntryValue(MI, DefinedRegs, OpenRanges, VarLocIDs);
     }
   }
 
+  if (AsyncDbgValues.size()) {
+    // Make sure the async entry values are at the very start.
```

Comment:
Could you leave a comment explaining why this is needed?
If the DBG_VALUEs are already in the entry MBB, but LDV drops them accidentally if they're not present at the very start, that could be a separate LDV bug to fix.

Reply:
The problem that the register allocator inserts a few copy instructions before the DBG_VALUEs, and normally that would be the correct thing to do, but for virtual backtraces the DBG_VALUEs must be on the DW_AT_low_pc of the function, or the variables aren't visible. And since they directly refer to the incoming register, that's also correct.

Reply:
And apparently that only happens on arm64.

Reply:
I see, so there's some separate, async-specific requirement that the locations for the async arguments start at the low_pc of the function/funclet?